### ISSUE TITLE
ID list cannot handle large page numbers

### ISSUE THREAD

**Issue by tristanr-cogapp** [ArchiveLabs/iiif.archivelab.org#17]
The list of IDs fails after a certain point, so only the first 10,000 can be retrieved, instead of the several million allegedly available.

Steps to recreate: use a "page" value higher than 100.

I.e. https://iiif.archivelab.org/iiif/?limit=100&page=100 <= works as expected

But https://iiif.archivelab.org/iiif/?limit=100&page=101 <= fails, with "error": "string indices must be integers"

Has the limit of 100 that applies to the "limit" parameter also been incorrectly applied to the "page" parameter?

**Comment by mekarpeles** [comment]
Thank you for reporting this @tristanr-cogapp. The problem is 3-layers deep:
1) Our IIIF service catalog relies on the experimental Archive Labs endpoint: https://api.archivelab.org/items
2) The Archive Labs experimental items endpoint calls archive.org/advancedsearch.php (under the hood)
3) The official archive.org Advanced Search service now limits queries to 10000 results
e.g. http://archive.org/advancedsearch.php?q=collection%3Alibrivoxaudio&fl[]=identifier&sort[]=&sort[]=&sort[]=&rows=100&page=1000&output=json&callback=callback&save=yes

"RANGE_OUT_OF_BOUNDS: paging is only supported through 10000 results; scraping is supported through the Scraping API, see https://archive.org/help/aboutsearch.htm  or, you may request up to 100000000 results at one time if you do NOT specify any page. For best results, do NOT specify sort (sort may be automatically disabled for very large queries)."

TL;DR, the api.archivelab.org/items endpoint should be updated to use the scraping API described in https://archive.org/help/aboutsearch.htm

This is high on my personal priority list, but I'm unable to prioritize it above my other responsibilities. Will keep the community posted.

**Comment by mekarpeles** [comment]
To unblock your use case, I've addressed this bug today although unfortunately it changes the interface of the API:

e.g. http://iiif.archivelab.org/iiif/?cursor=W3siaWRlbnRpZmllciI6IjAwMDU1MkFfMDAwMDQ5OTlfMF84NWJkNDkwMy01YzcxLTQwODEtOWI5Ni1lNjRjNDE2MWNkMWIifV0=

This uses the scrape API under the hood (instead of advancedsearch.php). Refer to  https://archive.org/help/aboutsearch.htm for documentation. A `cursor` value (which is returned with each request) enables you to request the next page or refer to the previous page.

If you're still blocked or want this endpoint to behave differently, please let us know.

**Comment by tristanr-cogapp** [comment]
Thanks for the prompt response @mekarpeles.

I have verified that the new API is working for large page values. However, there now seems to be a problem with almost all of the identifiers that it returns, which I have logged separately at https://github.com/ArchiveLabs/iiif.archivelab.org/issues/18
